client/src/pages/ViewWorkouts.jsx

```diff
@@ -6,42 +6,46 @@ import { Workout } from "../components/Workout"
 //redux components
 import { useDispatch, useSelector } from "react-redux"
 //prettier-ignore
-import {getWorkouts,deleteWorkout,reset} from "../features/workouts/workoutsSlice"
+import {getWorkouts,deleteWorkout, getWorkoutsByDate, reset} from "../features/workouts/workoutsSlice"
 import { removeUserData } from "../features/auth/authSlice"
 import { useEffect, useState } from "react"
 import Spinner from "../components/Spinner"
 import styles from "../components/styles/ViewWorkouts.module.css"
 
 export function ViewWorkouts() {
+	//when the user goes back to the view workouts page, the calendar shows the date they chose but the workouts aren't being shown any more. This is because the workouts by date are only retrieved when the date changes, so if we want the workouts to be shown when the user goes back, we have to make a request based on if there is a date parameter in the URL i think instead of onChange???? THE SOLUTION WAS TO MAKE A GET REQUEST OF THE WORKOUTS BY DATE IF THERE IS A DATE IN THE URL PARAMETER ALONG WITH PUTTING IT INSIDE OF A USEEFFECT THAT WAY THE WORKOUTS BY DATE LOAD ON HITTING THE BACK BUTTON AS USEEFFECT RUNS ONCE ON COMPONENT LOAD AND DEP. ARRAY VALUES CHANGING.
 	const navigate = useNavigate()
 	const { date } = useParams()
 	const [calendarDate, setCalendarDate] = useState(date)
 
 	const dispatch = useDispatch()
-	const [dateWorkouts, setDateWorkouts] = useState([])
 
 	const { userData } = useSelector((state) => state.auth)
 	const { workouts, isLoading, isError, message } = useSelector((state) => {
 		return state.workouts
 	})
 
 	//Capitalize First letter of first and last name. Will probably move this to happen elsewhere
```

Comment:
Why did you delete and write the same code again?

Reply:
I did this because I didn't like the code on multiple lines. It's easier to read now

Reply:
nj